src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -2289,6 +2284,13 @@ void AchievementGlobalMgr::LoadAchievementCriteriaList()
             }
         }
 
+        for (uint32 conditionField = 0; conditionField < MAX_CRITERIA_REQUIREMENTS; ++conditionField)
+        {
+            if (criteria->AdditionalRequirements[0].Type != ACHIEVEMENT_CRITERIA_CONDITION_NONE)
+                if (conditionField == 0 || criteria->AdditionalRequirements[conditionField].Type != criteria->AdditionalRequirements[conditionField-1].Type)
+                    m_AchievementCriteriasByCondition[criteria->AdditionalRequirements[conditionField].Type][criteria->AdditionalRequirements[conditionField].Asset].push_back(criteria);
```

Comment:
Add a helpful ASSERT on additional requirement type, just like the one for criteria types above